### ISSUE TITLE
CLI options requiring equals sign not passed correctly

### ISSUE THREAD

**Issue by horsepress** [75lb/handbrake-js#16]
Some Handbrake CLI options, such as "rotate", require to be passed with an equals sign, like this:

--rotate=4 

[for 90 degree rotataion]

At present this module passes them like this:

--rotate 4

Meaning that the default rotate setting is applied ("--rotate" , equivalent to "--rotate=3")

This is based on an option specified like this in the options object:

{
"rotate": 4
}

Passing it like this works:

{
"rotate=4" :""
}

However this is inconvenient when setting the value dynamically.

Would it be possible to flag the options that require the "=" sign  and to insert it in the CLI arguments as necessary? This might require a modification to the object-to-spawn-args.js module.

==========

The below CLI parameters illustrate the difference (some parameters go like "--paramname value" , others like "--paramname=value")

-i "IMG_1152.MOV" -t 1 --angle 1 -c 1 -o "test.mp4"  -f mp4  -w 720 --crop 0:0:0:0 --loose-anamorphic  --modulus 2 -e x264 -q 24 --vfr -a 1 -E av_aac -6 dpl2 -R Auto -B 160 -D 0 --gain 0 --audio-fallback ac3 --markers="test-1-chapters.csv" --encoder-preset=veryfast  -x  , --rotate=4 --encoder-level="4.0"  --encoder-profile=main  --verbose=1

**Comment by horsepress** [comment]
A crude fix is to insert this in Handbrake.js, in Handbrake.prototype._run, round about line 66 (before ```this._emitStart()``` ):

```javascript
  ['rotate','encoder-preset','encoder-level','encoder-profile','markers'].forEach(function(propertyName){
	if (this.options[propertyName]){
		var propNameWithEqualsValue = propertyName + "=" + this.options[propertyName];
		this.options[propNameWithEqualsValue] = '';
		delete this.options[propertyName];
	}
  },this);
```
 
(there are probably more options that require equals than I've put in the array).

**Comment by 75lb** [comment]
hmm, i see.. thanks for alerting me to these rogue options.. if you submit your PR i will apply it while we decide what to do long term..

**Comment by 75lb** [comment]
tbh, this being an anomaly unique to handbrake, a patch like yours correcting irregularities might be the best we can do.. i don't see the point in making this a permanent feature of object-to-spawn-args..

**Comment by 75lb** [comment]
fixed and released in v2.0.9
